app/src/common/shared/com/igalia/wolvic/utils/DictionariesManager.java

```diff
@@ -0,0 +1,153 @@
+package com.igalia.wolvic.utils;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.igalia.wolvic.downloads.Download;
+import com.igalia.wolvic.downloads.DownloadJob;
+import com.igalia.wolvic.downloads.DownloadsManager;
+import com.igalia.wolvic.ui.widgets.WidgetManagerDelegate;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Objects;
+
+public class DictionariesManager implements DownloadsManager.DownloadsListener {
+
+    static final String LOGTAG = SystemUtils.createLogtag(DictionariesManager.class);
+
+    private final Context mContext;
+    private final DownloadsManager mDownloadManager;
+    private long mDicDownloadLang = -1;
+
+    public DictionariesManager(@NonNull Context context) {
+        mContext = context;
+        WidgetManagerDelegate mApplicationDelegate = ((WidgetManagerDelegate) context);
+        mDownloadManager = mApplicationDelegate.getServicesProvider().getDownloadsManager();
+    }
+
+    public void init() {
+        mDownloadManager.addListener(this);
+    }
+
+    public void end() {
+        mDownloadManager.removeListener(this);
+    }
+
+    @Nullable
+    public String getOrDownloadDictionary(@NonNull String lang) {
```

Comment:
Looks like the return value is never used. Can we get rid of it? Or where you thinking in other use cases were it could be useful?

Reply:
Let's just remove the return value here.